src/promises/services.py

```diff
@@ -1,79 +1,52 @@
-                    # We go through the case subject ids and find any common subjects
-                    # in the promise subject id array
-                    for case_subject_id in case_subject_ids:
-                        CRONLOGGER.info('case subject id: '+str(case_subject_id))
-                        CRONLOGGER.info('promise subject id: '+str(promise_subject_ids))
-                        CRONLOGGER.info(case_subject_id in promise_subject_ids)
-                        if case_subject_id in promise_subject_ids:
-                            number_of_common_subjects += 1
-
-                    CRONLOGGER.info(number_of_common_subjects)
-                    CRONLOGGER.info(size_of_larger_subject_array)
-
-                    # If the percent between common subjects and the largest array
-                    # reaches a certain point we want to connect the case and promise. If
-                    number_of_common_subjects_float = float(number_of_common_subjects)
-                    size_of_larger_subject_array_float = float(size_of_larger_subject_array)
-                    percent_of_common_subjects = number_of_common_subjects_float / size_of_larger_subject_array_float
-                    CRONLOGGER.info(percent_of_common_subjects)
-                    # We want at least 4 subjects on both the case and the promise
-                    if len(case_subject_ids) > 4 and len(promise_subject_ids) > 4:
-                        CRONLOGGER.info('case and promise have more than 4 subjects each')
-                        # If the percent is 80% or higher we make a connection
-                        if percent_of_common_subjects >= 0.8:
-                            PromiseCase.objects.create(case=case, promise=promise)
-                            CRONLOGGER.info('PromiseCase connection created')
-                        # If the percent is between 50% and 79% we create a suggested connection
-                        elif percent_of_common_subjects >= 0.5:
-                            SuggestedPromiseCase.objects.create(case=case, promise=promise)
-                            CRONLOGGER.info('SuggestedPromiseBill connection created')
+    current_parliament = Parliament.objects.latest('created')
+    parliament_sessions_ids = ParliamentServices.parliament_session_ids_by_parliament(current_parliament)
+    current_promises = Promise.objects.filter(parliament=current_parliament.id) 
+    cases_in_sessions = Case.objects.filter(parliament_session__in=parliament_sessions_ids)
+
+    for case in cases_in_sessions:
+        case_subject_ids = CaseSubject.objects.filter(case=case).values_list('subject_id', flat=True)
+        
+        # We look at each promise within the same parliament
+        for promise in current_promises:
+            promise_subject_ids = PromiseSubject.objects.filter(promise=promise).values_list('subject_id', flat=True)
+            if len(promise_subject_ids) > 4:
+                # We want more than 4 subjects on the promise
```

Comment:
Maybe a small note on _why_ 5 is the lowest number?
Even though it's probably subject to change